### PR TITLE
Added the flag and added tests

### PR DESCRIPTION
**Pull request by kjawadDeveloper2**
1) Added the flag in the recipe of easel.
2) Added the test cases for the showing of buy not button logic. 
3) Removed dead code.

### CODE REVIEW

File: wallet/lib/pages/purchase_item/purchase_item_view_model.dart [hunk 1]
```diff
@@ -454,15 +435,49 @@ class PurchaseItemViewModel extends ChangeNotifier {
...
+      return true;
+    }
+
+    return isIOSStripePaymentAllowed(isPlatformAndroid: isPlatformAndroid);
   }
 
   NFT get nft => _nft;
 
-  final ValueNotifier<bool> shouldShowBuyNow = ValueNotifier(false);
+  // final ValueNotifier<bool> shouldShowBuyNow = ValueNotifier(false);
```
Comment (mustafapylons) [inline wallet/lib/pages/purchase_item/purchase_item_view_model.dart:480]: I think this dead code can be removed as well.

**Review by mustafapylons — COMMENTED** [review]
One small comment, otherwise code is good.

File: wallet/lib/pages/purchase_item/purchase_item_view_model.dart (hunk 1 above)
Comment (kjawadDeveloper) [reply wallet/lib/pages/purchase_item/purchase_item_view_model.dart:480]: yes sure.

**Comment by kjawadDeveloper2** [comment]
https://github.com/Pylons-tech/pylons/issues/931

**Review by mustafapylons — APPROVED** [review]
Looks good to me.

**Review by MikeSofaer — APPROVED** [review]

**Pull request merged**